src/operators.jl

```diff
@@ -501,13 +501,45 @@ for op in [:+, :-, :*, :/]; @eval begin
     end
 end; end
 
-# Special-case sparse matrix scalar multiplicaiton
+for op in [:*, :/]; @eval begin
+    function $op{T<:JuMPTypes}(lhs::Number,rhs::Array{T})
```

Comment:
```jl
m = Model()
@defVar(m, x)
A = SparseMatrixCSC(3, 3, [1:4;], [1:3;], [x,x,x])
B = 3A
```
You want ``B`` to be sparse as well, which ``similar`` gives you, but then ``eachindex`` does iteration over all elements, not just nonzeros.